zerver/lib/upload.py

```diff
@@ -33,10 +35,8 @@
 # slugify.
 
 def sanitize_name(name):
-    split_name = name.split('.')
-    base = ".".join(split_name[:-1])
-    extension = split_name[-1]
-    return slugify(base) + "." + slugify(extension)
+    fileName = force_text(name).strip().replace(' ', '_')
+    return re.sub(r'(?u)[^-\w.]', '', fileName)
```

Comment:
@timabbott Sorry for the late response. I was occupied in other works. Yes that is very reasonable. I'll try to make it based on slugify. I'll also provide a documentation after changing.